src/posix.c

```diff
@@ -62,8 +62,11 @@ int p_getaddrinfo(
 	ai = ainfo;
 
 	for (p = 1; ainfo->ai_hostent->h_addr_list[p] != NULL; p++) {
-		ai->ai_next = malloc(sizeof(struct addrinfo));
-		memcpy(&ai->ai_next, ainfo, sizeof(struct addrinfo));
+		if (!(ai->ai_next = malloc(sizeof(struct addrinfo)))) {
```

Comment:
It's not 100% expected that the `p_*` functions that we write should set library error codes.  These are often used as fallbacks when an actual posix function is not available.  And - of course - the actual posix functions (like the system's `getaddrinfo`) will never set the library error codes.

Therefore `p_getaddrinfo` doesn't need to set the library error code.  Instead, it needs to set something that `p_gai_strerror` can cope with (which is our reimplementation of `gai_strerror`.  [Indeed that does so - `-1` is treated as an OOM scenario.](https://github.com/libgit2/libgit2/blob/a8d0b4de92b83745308b2d9af0f62828bfb550a1/src/posix.c#L94)

As a result, I think this change is correct.